axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/ProjectBusinessServiceImpl.java

```diff
@@ -761,4 +764,31 @@ protected String getProgressIcon(int comparisonResult) {
         return "";
     }
   }
+
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public void transitionBetweenPaidStatus(Project project) throws AxelorException {
+    ProjectStatus completedStatus = appProjectService.getCompletedProjectStatus();
+    ProjectStatus completedPaidStatus =
+        appProjectService.getAppProject().getCompletedPaidProjectStatus();
+    if (Objects.isNull(completedPaidStatus)) {
+      throw new AxelorException(
+          appProjectService.getAppProject(),
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(ProjectExceptionMessage.PROJECT_CONFIG_COMPLETED_PAID_PROJECT_STATUS_MISSING));
+    }
```

Comment:
it's better to factorize this code also in appProjectService, also please write `== null` to check if something is `null`.